### PR TITLE
Fixes #13258 - Use patternfly icons for parameters

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

File: app/helpers/layout_helper.rb [hunk 1]
```diff
@@ -353,15 +353,16 @@ def base_errors_for(obj)
...
-    content_tag(:a, icon_text(options[:icon], title), { :rel => "popover",
-                                                        :data => { :content => msg,
-                                                                   :"original-title" => title,
-                                                                   :trigger => "focus",
-                                                                   :container => 'body',
-                                                                   :html => true },
-                                                        :role => 'button',
-                                                        :tabindex => '-1' }.deep_merge(options))
+    options[:icon] ||= 'info-circle'
+    options[:kind] ||= 'fa'
```
Comment (dLobatog) [inline app/helpers/layout_helper.rb]: Default to kind `pficon`, icon `info` instead, that way it looks more homogeneous (see the status icons for instance)

File: app/views/puppetclasses/_form.html.erb [hunk 2]
```diff
@@ -45,11 +45,11 @@
           </div>
         </div>
         <div class="smart-var-left col-md-3">
-          <ul class="nav nav-pills nav-stacked smart-var-tabs" data-tabs="pills">
+          <ul class="nav nav-pills nav-stacked smart-var-tabs" data-tabs="pills">yeah
```
Comment (dLobatog) [inline app/views/puppetclasses/_form.html.erb]: huh?

File: app/views/puppetclasses/_form.html.erb [hunk 3]
```diff
@@ -45,11 +45,11 @@
...
             <% @puppetclass.class_params.includes(:environments, :environment_classes, :lookup_values).each do |key| %>
               <li data-used-environments=<%= key.environments.map(&:to_s).to_json %> >
                 <a data-toggle="tab" id="pill_<%= key.to_param %>" href="#<%= key.to_param %>" title="<%= key %>">
-                  <div class="clip"><%= icon_text((key.override ? "flag": ""), key.to_s.tr('_',' ')) %></div>
+                  <div class="clip"><%= icon_text((key.override ? "flag": ""), key.to_s.tr('_',' '), :kind => 'fa') %></div>
```
Comment (dLobatog) [inline app/views/puppetclasses/_form.html.erb:52]: `:kind pficon`

File: app/views/puppetclasses/_form.html.erb (hunk 2 above)
Comment (orrabin) [inline app/views/puppetclasses/_form.html.erb]: test leftover :)

File: app/helpers/lookup_keys_helper.rb [hunk 4]
```diff
@@ -110,38 +111,38 @@ def lookup_key_description(lookup_key, matcher, inherited_value)
...
 
-  def hidden_toggle(hidden, hide_icon = 'eye-close', unhide_icon = 'eye-open', strikethrough = false)
+  def hidden_toggle(hidden, hide_icon = 'eye-slash', unhide_icon = 'eye', strikethrough = false)
     return unless can_edit_params?
     if strikethrough && !hidden
-      link_to_function(icon_text(hide_icon), "", :class =>"btn btn-default btn-md btn-hide disabled")
+      link_to_function(icon_text(hide_icon, '', :kind => 'fa'), "", :class =>"btn btn-default btn-md btn-hide disabled")
     else
-      link_to_function(icon_text(unhide_icon), "input_group_hidden(this)",
+      link_to_function(icon_text(unhide_icon, '', :kind => 'fa'), "input_group_hidden(this)",
```
Comment (dLobatog) [inline app/helpers/lookup_keys_helper.rb:141]: Can we use the same eye-slash icon on Puppet class parameters? The 'A' we're using is less clear.

File: app/helpers/lookup_keys_helper.rb [hunk 5]
```diff
@@ -110,38 +111,38 @@ def lookup_key_description(lookup_key, matcher, inherited_value)
   end
 
   def lookup_key_warnings(required, has_value)
-    return { :text => '', :icon => 'info-sign' } if has_value
+    return { :text => '', :icon => 'info-circle' } if has_value
```
Comment (dLobatog) [inline app/helpers/lookup_keys_helper.rb]: pficon-info

File: app/helpers/lookup_keys_helper.rb [hunk 6]
```diff
@@ -110,38 +111,38 @@ def lookup_key_description(lookup_key, matcher, inherited_value)
...
 
     if required
       { :text => _("Required parameter without value.<br/><b>Please override!</b><br/>"),
-        :icon => "warning-sign" }
+        :icon => "exclamation-triangle" }
```
Comment (dLobatog) [inline app/helpers/lookup_keys_helper.rb]: warning-triangle-o

File: app/helpers/lookup_keys_helper.rb [hunk 7]
```diff
@@ -110,38 +111,38 @@ def lookup_key_description(lookup_key, matcher, inherited_value)
...
     else
       { :text => _("Optional parameter without value.<br/><i>Will not be sent to Puppet.</i><br/>"),
-        :icon => "exclamation-sign" }
+        :icon => "exclamation-circle" }
```
Comment (dLobatog) [inline app/helpers/lookup_keys_helper.rb]: warning-triangle-o

File: app/views/puppetclasses/_form.html.erb (hunk 3 above)
Comment (orrabin) [inline app/views/puppetclasses/_form.html.erb:52]: I didn't like the 'pficon-flag' icon, the 'fa-flag' is closer to what we have today.

File: app/helpers/lookup_keys_helper.rb [hunk 8]
```diff
@@ -97,7 +97,8 @@ def diagnostic_popover(lookup_key, matcher, inherited_value, warnings)
     popover('', description.prepend(warnings[:text]),
             :data => { :placement => 'top' },
             :title => _("Original value info"),
-            :icon => warnings[:icon])
+            :icon => warnings[:icon],
+            :kind => 'fa')
```
Comment (dLobatog) [inline app/helpers/lookup_keys_helper.rb]: `kind => pficon`

File: app/helpers/layout_helper.rb [hunk 9]
```diff
@@ -486,7 +487,7 @@ def last_days(days)
 
   def fullscreen_button(element = "$(this).prev()")
     button_tag(:type => 'button', :class => 'btn btn-default btn-md btn-fullscreen', :onclick => "set_fullscreen(#{element})", :title => _("Full screen")) do
-      icon_text('resize-full')
+      icon_text('expand', '', :kind => 'fa')
```
Comment (dLobatog) [inline app/helpers/layout_helper.rb:490]: `kind => pficon`

File: app/assets/javascripts/lookup_keys.js [hunk 10]
```diff
@@ -148,7 +148,7 @@ function toggleOverrideValue(item) {
 
   fields_to_disable.prop('disabled', !override);
   default_value_field.prop('disabled', !override || use_puppet_default);
-  override ? pill_icon.addClass('glyphicon-flag') : pill_icon.removeClass('glyphicon-flag');
+  override ? pill_icon.addClass('fa-flag') : pill_icon.removeClass('fa-flag');
```
Comment (dLobatog) [inline app/assets/javascripts/lookup_keys.js:151]: `pficon-flag`

File: app/helpers/lookup_keys_helper.rb (hunk 4 above)
Comment (orrabin) [inline app/helpers/lookup_keys_helper.rb:141]: This is intentional to distinguish this from the parameters that actually change the hidden flag from here. The decision to use a different icon was agreed on after discussing this with Kyle.

File: app/helpers/lookup_keys_helper.rb (hunk 6 above)
Comment (orrabin) [inline app/helpers/lookup_keys_helper.rb]: Should the required and optional parameters have the same sign? How about:  pficon-error-circle-o?

File: app/helpers/layout_helper.rb (hunk 9 above)
Comment (orrabin) [inline app/helpers/layout_helper.rb:490]: I didn't find the same icon we had for full screen in pficon so I used this one.

**Comment by orrabin** [comment]
@dLobatog I fixed some of the commits and responded to the others.

File: app/helpers/layout_helper.rb (hunk 9 above)
Comment (dLobatog) [inline app/helpers/layout_helper.rb:490]: Nevermind, I overlooked this

File: app/helpers/common_parameters_helper.rb [hunk 11]
```diff
@@ -41,7 +41,7 @@ def use_puppet_default_help link_title = nil, title = _("Use Puppet default")
 
   def hidden_value_field(f, field, value, disabled, options = {})
     if f.object.hidden_value?
-      input = f.password_field(field, :disabled => disabled, :value => value, :class => 'form-control')
+      input = f.password_field(field, :disabled => disabled, :value => value, :class => 'form-control no-stretch')
```
Comment (dLobatog) [inline app/helpers/common_parameters_helper.rb:44]: This conflicts with the hidden params PR I just merged, I'll fix it when I merge this one.

**Comment by dLobatog** [comment]
Merged as ae23d00776eacd6cb37ecf68e8201a846896eb07, I took care of a tiny conflict, Jenkins was green - http://ci.theforeman.org/job/test_develop_pull_request/14118/

**Comment by dLobatog** [comment]
Thanks @orrabin !

**Pull request closed without merging**